bot/glossary_command.go

```diff
@@ -0,0 +1,149 @@
+package bot
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/quintilesims/slackbot/db"
+	"github.com/quintilesims/slackbot/models"
+	glob "github.com/ryanuber/go-glob"
+	"github.com/urfave/cli"
+)
+
+// NewGlossaryCommand returns a cli.Command that manages !glossary
+func NewGlossaryCommand(store db.Store, w io.Writer) cli.Command {
+	return cli.Command{
+		Name:  "!glossary",
+		Usage: "manage the glossary",
+		Subcommands: []cli.Command{
+			{
+				Name:      "define",
+				Usage:     "add or set an entry in the glossary",
+				ArgsUsage: "KEY DEFINITION",
+				Action:    newGlossaryDefineAction(store, w),
+			},
+			{
+				Name:      "rm",
+				Usage:     "remove an entry from the glossary",
+				ArgsUsage: "KEY",
+				Action:    newGlossaryRemoveAction(store, w),
+			},
+			{
+				Name:      "search",
+				Usage:     "search for entries in the glossary",
+				ArgsUsage: "GLOB",
+				Action:    newGlossarySearchAction(store, w),
+			},
+		},
+	}
+}
+
+func newGlossaryDefineAction(store db.Store, w io.Writer) func(c *cli.Context) error {
+	return func(c *cli.Context) error {
+		args := c.Args()
+		key := args.First()
+		definition := strings.Join(args.Tail(), " ")
+
+		if key == "" {
+			return fmt.Errorf("Arg KEY is required")
+		}
+
+		if definition == "" {
+			return fmt.Errorf("Arg DEFINITION is required")
+		}
+
+		glossary := models.Glossary{}
+		if err := store.Read(db.GlossaryKey, &glossary); err != nil {
+			return err
+		}
+
+		var text string
+		if previousDefinition, ok := glossary[key]; previousDefinition != "" && ok {
```

Comment:
I don't think we need this; let's just set `glossary[key] = definition` and not worry about it. Makes the code look quite a bit cleaner imo.